calyx-opt/src/passes/top_down_compile_control.rs

```diff
@@ -219,9 +222,23 @@ struct Schedule<'b, 'a: 'b> {
     pub builder: &'b mut ir::Builder<'a>,
 }
 
+#[derive(PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+struct FSMInfo {
+    pub component: String,
+    pub group: String,
+    pub states: Vec<FSMStateInfo>,
+}
+
+#[derive(PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+struct FSMStateInfo {
+    id: u64,
+    group: String,
```

Comment:
Store `Id` not `String`.

Reply:
I've been struggling with a compiler error saying that the trait `Serialize` is not implemented for `Id`, even though there is a `impl` for `serde::Serialize` in `Id` ([here](https://github.com/calyxir/calyx/blob/main/calyx-utils/src/id.rs#L96)). I've tried playing around with the attributes in `id.rs` but still ran into compilation issues... Do you mind pointing me to how I can fix this problem, or what to check?

Reply:
It's feature gated so one reason it might not be visible is the `cfg`. Can you try removing it and seeing if that helps?

Reply:
I removed the `cfg` and added `use serde::{Serialize, Serializer};` but I got an error saying " use of undeclared crate or module `serde`" because `serde` is optional under `calyx-utils`, but making it non-optional breaks `calyx-ir` and so on (because they depend on `calyx-utils`.

I'll ask around either on slack or in person about this, hopefully I'll resolve this soon!

Reply:
Fixed this with @EclecticGriffin ! :)